json-rpc/docs/method_get_transactions_with_proofs.md

```diff
@@ -2,29 +2,34 @@
 
 **Description**
 
-Get the transactions on the blockchain along with the proofs necessary to verify the said transactions.
+Get the transactions on the blockchain along with the proofs necessary to verify said transactions.
 
 ### Parameters
 
-| Name           | Type           | Description                                                          |
-|----------------|----------------|----------------------------------------------------------------------|
-| start_version  | unsigned int64 | Start on this transaction version for this query                     |
-| limit          | unsigned int64 | Limit the number of transactions returned, the max value is 1000     |
+| Name           | Type           | Description                                                           |
+|----------------|----------------|-----------------------------------------------------------------------|
+| start_version  | unsigned int64 | Start on this transaction version for this query                      |
+| limit          | unsigned int64 | Limit the number of transactions returned, the max value is 1000      |
+| include_events | boolean        | When set to true, also include serialized events for each transaction |
 
 ### Returns
 
-| Name                      | Type               | Description                   |
-|---------------------------|--------------------|-------------------------------|
-| serialized_transactions   | List<string>       | An array of hex encoded strings with the raw bytes of the returned `Transaction` |
-| proofs                    | TransactionsProofs | The proofs, see below.   |
+If include_events is false, the serialized_events field will not be included.
+
+| Name                      | Type                   | Description                                                                                                 |
+|---------------------------|------------------------|-------------------------------------------------------------------------------------------------------------|
+| serialized_transactions   | List<string>           | An array of hex encoded strings with the raw bytes of the returned `Transaction`                            |
+| serialized_events         | Option<string>         | An optional BCS-serialized list of each transaction's list of events or `null` if `include_events` is false |
+| first_transaction_version | Option<unsigned int64> | If there are any transactions, then the version of the first transaction, otherwise `null`                  |
```

Comment:
why we need this? isn't the `start_version` in the parameter is the first transaction version?

Reply:
I could go either way. On the one hand, I'd prefer the *_with_proofs APIs to avoid trying to be smart and instead just expose the underlying storage API without changes. On the other hand, we can save some bytes and remove this field which the caller can reconstitute. I definitely lean toward "no changes", but if other people have stronger opinions I can remove this field.

Reply:
I think underline APIs are free to change, json-rpc API response should be more restricted to make sure it can be as stable as possible.
For this field, the question is not what it is, the question is why it is different with the `start` param. If they are never different, why we output it.

Reply:
Makes sense, I'll remove this field 👍